### PR TITLE
Allows tasks to block dispatch

### LINKED ISSUE #14056: Allow tasks to block dispatch
Placeholder to implement changes detailed in tech spec written in #14055

### PR DESCRIPTION
**Pull request by lomky**
Resolves #14056 

Part 1 of the stack:

Part 1: this PR
Part 2: #15041 
Part 3: #15042 

### Description
Renames 'blocking?' to 'blocking_distribution?'
Implements 'blocking_dispatch' and applies it to [appropriate tasks](https://github.com/department-of-veterans-affairs/caseflow/issues/14260#issue-616982555)

### Acceptance Criteria
- [ ] Full Stack: @lomky and @hschallhorn have doublechecked our enable reasoning
- [ ] Tests pass!
- [ ] Code compiles correctly
- [ ] Certain tasks will have the ability to block dispatch
   - [ ] CongressionalInterestMailTask
   - [ ] FoiaRequestMailTask
   - [ ] PrivacyActRequestMailTask
   - [ ] PrivacyComplaintMailTask
   - [ ] FoiaColocatedTask
   - [ ] FoiaTask
   - [ ] PrivacyActTask
- (Stack 2) These task should be created under RootTask* or BvaDispatchTask (Stack 2)
- (Stack 3) BvaDispatch tasks are not created when tasks that block dispatch are still open (Stack 3)
- (Stack 3) BvaDispatchTask is created when all tasks that block dispatch are closed (Stack 3)

### Testing Plan
1. Load up the console!
2. Create a direct review appeal at ready for distribution
3. Create Tasks of the types listed in the AC - the **should not** be created under the DistributionTask
   - [ ] Dispatch blocking tasks do not block Distribution
4. Create a few tasks that should still block distribution - the **should** be created under the Distribution Task
   - [ ] Distribution blocking tasks block Distribution

- Spend a bit of time grepping through the codebase with various permutations of the word 'blocking'
   - [ ] places that say 'blocking' now specify 'blocking distribution' or 'blocking dispatch'
   - [ ] places testing for blocking distribution properly invoke a task that still blocks distribution

### CODE REVIEW

File: spec/models/task_spec.rb [hunk 1]
```diff
@@ -1700,4 +1700,52 @@ def next_assignee(_options = {})
...
+        end
+      end
+    end
+
+    context "on legacy appeals, a normally blocking task" do
+      let(:appeal) { create(:legacy_appeal) }
+      let(:task) { create(:congressional_interest_mail_task, appeal: appeal) }
+
+      it "does not block dispatch" do
+        expect(task.blocking_dispatch?).to be(false)
```
Comment (hschallhorn) [inline spec/models/task_spec.rb:1733]: 😤

File: app/models/tasks/foia_task.rb [hunk 2]
```diff
@@ -4,6 +4,10 @@
 # Task to track when a Freedom of Information Act task has been assigned to Privacy Team
 
 class FoiaTask < Task
+  def self.blocking_dispatch?
+    FeatureToggle.enabled?(:block_at_dispatch) ? true : false
+  end
```
Comment (hschallhorn) [inline app/models/tasks/foia_task.rb:9]: Is this needed as it already exists on the FoiaColocatedTask? Wouldn't creating a foia colocated task result in the colocated task being made off the root task, as well as the foia task being created on the root task?

Comment (lomky) [reply app/models/tasks/foia_task.rb:9]: my worry is that while the only way to create a FoiaTask today is via a FoiaColocatedTask, there's nothing in our larger process that guarantees that?

File: app/models/tasks/congressional_interest_mail_task.rb [hunk 3]
```diff
@@ -1,8 +1,12 @@
...
 
 class CongressionalInterestMailTask < MailTask
-  def self.blocking?
-    true
+  def self.blocking_distribution?
+    FeatureToggle.enabled?(:block_at_dispatch) ? false : true
+  end
+
+  def self.blocking_dispatch?
+    FeatureToggle.enabled?(:block_at_dispatch) ? true : false
```
[the remark this answers is not in the document]
Comment (lomky) [reply app/models/tasks/congressional_interest_mail_task.rb:9]: hah, nope

File: app/models/tasks/foia_task.rb (hunk 2 above)
Comment (hschallhorn) [reply app/models/tasks/foia_task.rb:9]: we could create a check on the parent as we do with [transcription tasks](https://github.com/department-of-veterans-affairs/caseflow/blob/ee6a5a7876e952bf9b453e0bb376b0e3482fb8a8/app/models/tasks/transcription_task.rb#L24-L25)

**Pull request closed without merging**